Fixing support for KiSAO

jonrkarr opened the pull request:
Fixing support for KiSAO:
- Applying new KiSAO term KISAO_0000559 (initial step size); closes #484 
- Applying new KISAO term KISAO_0000569 (NLEQ2); closes #486 
- Adding support for correct KiSAO ids (with pattern `KISAO_\d{7}`); closes #488 

These changes maintain backward support:
- Old semantically different step size term KISAO:0000332
- Old non-specific newton-type terms
- Old incorrect KiSAO id pattern `KISAO:\d{7}`

luciansmith commented:
This is a fine change, but we also need something to maintain backwards compatibility:  we should continue to read files with 0000332 as meaning 'initial time step'.

jonrkarr commented:
I think continuing to recognize `0000332` is fine. tellurium already does something similar for algorithms -- it recognizes several terms that are distant synonyms for the particular algorithms that it supports. (I understand why you're doing this. It makes sense given that tellurium is one of the very few tools that supports KiSAO. I think a better way to approach this is to use the KiSAO hierarchy to recommend alternative terms rather than trying to embed this ad hoc into each simulation tool.)

Are there files that use this term? From my vantage support, usage of KiSAO is pretty minimal. At this point, it seems that there's little need to provide backwards compatibility for files that don't exist.

luciansmith commented:
I agree that something that actually understands the KiSAO hierarchy would be great, if we get the time to do it.

As far as existing files, I really have no idea--tellurium has a wide user base, but I personally don't know what features they tend to use.  In general, I just don't want to break someone's pipeline they might have set up for themselves or their lab (especially if it doesn't take that much to continue to be backwards compatible).

jonrkarr commented:
I added a line to maintain backwards compatibility with the semantically different term `KISAO_0000332`.

I changed tellurium to recognize the correct KiSAO ids (with pattern `KISAO_\d{7}`), as well as maintain backward support for the incorrect id pattern `KISAO:\d{7}`.

These changes will be necessary for inclusion in BioSimulators.

jonrkarr commented:
I also added a TODO to correct the generation of SED-ML.

In `tesedml.py`, the following 
```
algorithm.setKisaoID("KISAO:0000019")
```

should be replaced with this
```
algorithm.setKisaoID("KISAO_0000019")
```

luciansmith commented:
In the end, I took this PR as a template and implemented support for the new KiSAO terms in a slightly different way.  But the support is now in there, and thank you again for the code!
